src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -372,10 +393,29 @@ private HostSecrets GenerateHostSecrets()
             return PersistSecretsAsync(refreshedSecrets, keyScope);
         }
 
-        private Task PersistSecretsAsync<T>(T secrets, string keyScope = null) where T : ScriptSecrets
+        private async Task PersistSecretsAsync<T>(T secrets, string keyScope = null, bool isNonDecryptable = false) where T : ScriptSecrets
         {
+            ScriptSecretsType secretsType = secrets.SecretsType;
+            if (isNonDecryptable)
+            {
+                string prefix = keyScope ?? "host";
+
+                string[] secretBackups = await _repository.ListSecretsByPrefixAsync($"{prefix}.{ScriptConstants.NonDecryptable}");
+
+                if (secretBackups.Length >= ScriptConstants.MaximumSecretBackupCount)
+                {
+                    string message = string.Format(Resources.ErrorTooManySecretBackups, ScriptConstants.MaximumSecretBackupCount, prefix);
+                    _traceWriter.Verbose(message);
+                    _logger?.LogDebug(message);
+                    throw new InvalidOperationException(message);
+                }
+
+                string timeStamp = DateTime.UtcNow.ToString("yyyy-MM-dd HH.mm.ss.ffffff");
+                keyScope = $"{prefix}.{ScriptConstants.NonDecryptable}.{timeStamp}.json";
+                secretsType = ScriptSecretsType.Function;
```

Comment:
This logic is a bit unclear. Why is the secret type set to Function? We're also overloading the `keyScope` here, and passing a full file name (with an extension) to the repository. The logic for format file names should be part of the repository instead of the secret manager.